packages/odyssey-react-mui/src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,108 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef, useState, useMemo } from "react";
+import type { ChangeEvent, MouseEvent } from "react";
+import type { OutlinedInputProps, TooltipProps } from "@mui/material";
+import {
+  Tooltip,
+  IconButton,
+  Box,
+  InputLabel,
+  OutlinedInput,
+  InputAdornment,
+} from "@mui/material";
+import { Visibility, VisibilityOff } from "@mui/icons-material";
+import { useOid } from "../../utils";
+
+interface State {
+  password: string;
+  showPassword: boolean;
+}
+
+export interface PasswordInputProps
+  extends Omit<OutlinedInputProps, "type" | "label" | "defaultValue"> {
+  defaultValue?: string;
+  label: string;
+  tooltipLabel?:
+    | TooltipProps["title"]
+    | ((isHidden: boolean) => TooltipProps["title"]);
+}
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
+  (props, ref) => {
+    const {
+      tooltipLabel,
+      id,
+      label,
+      defaultValue: password = "",
+      ...rest
+    } = props;
+
+    const [values, setValues] = useState<State>({
+      password,
+      showPassword: false,
+    });
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+      setValues({ ...values, password: event.target.value });
+      props.onChange?.(event);
+    };
+
+    const handleClickShowPassword = () => {
+      setValues({
+        ...values,
+        showPassword: !values.showPassword,
+      });
+    };
+
+    const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+    };
+
+    const tooltipTitle = useMemo(() => {
+      return typeof tooltipLabel === "function"
+        ? tooltipLabel(values.showPassword === false)
+        : tooltipLabel;
+    }, [values, tooltipLabel]);
+
+    const oid = useOid(id);
+
+    return (
+      <Box>
```

Comment:
Rather than `<Box>`, should we use `<FormControl>` here to ensure we pick up all of the field context? MUI seems to use it in all of their examples.

https://mui.com/material-ui/api/form-control/